speeduino/idle.ino

```diff
@@ -592,7 +592,11 @@ void idleControl(void)
     
         idle_cl_target_rpm = (uint16_t)currentStatus.CLIdleTarget * 10; //Multiply the byte target value back out by 10
         if( BIT_CHECK(LOOP_TIMER, BIT_TIMER_1HZ) ) { idlePID.SetTunings(configPage6.idleKP, configPage6.idleKI, configPage6.idleKD); } //Re-read the PID settings once per second
-        if((currentStatus.RPM - idle_cl_target_rpm > configPage2.iacRPMlimitHysteresis*10) || (currentStatus.TPS > configPage2.iacTPSlimit)){ //reset integral to zero when TPS is bigger than set value in TS (opening throttle so not idle anymore). OR when RPM higher than Idle Target + RPM Histeresis (coming back from high rpm with throttle closed)
+
+        //This keep idle valve open using the OL table value when TPS or RPM is bigger than the hysteresis value or when DFCO (like a simple dashpot). And even resets the PID integral. Is for preventing RPM dips when coming back to idle from part throttle or even WOT
+        if ( (currentStatus.RPM > (idle_cl_target_rpm + configPage2.iacRPMlimitHysteresis*10) ) || (currentStatus.TPS > configPage2.iacTPSlimit) || lastDFCOValue )
+        {
+          idle_pid_target_value = FeedForwardTerm;
```

Comment:
Why are you changing the PID target? That will cause the PID to overshoot depending on the open loop value!

Reply:
In fact, it will override the PID value, forcing the idle valve to use the idle table but is just when going back to idle. That's why its cheking if the RPM is bigger than the idle + hysteresis or the TPS is above the TPS hysteresis. its like the megasquirt idle initial value

Reply:
No, that variable is the RPM target, the PID value is override with the feef forward value been passed as argument on the 'compute' call.
That change is terrible, you should remove it.

I understand the checks, you know, just that line makes no sense at all.

Reply:
ok, but i tested in my car and it work good, at least the stepper motor part

Reply:
Again, that don't means it will be OK for everyone, and after removing that like it will remains the same, just less chance to be a bug.

Reply:
Olha eu olhei a biblioteca que a speeduino usa para o PID  e  o idle_pid_target_value é na verdade a saída do PID que vai para o controle da válvula e  o idle_cl_target_rpm é o alvo em RPM que vem da tabela de alvo da marcha lenta .

![PID](https://github.com/noisymime/speeduino/assets/143891251/44fa887c-3162-4016-8a6a-10321b9aba95)

Reply:
Mesmo assim essa linha é desnecessária, acredite, eu refiz o PID todo e também escrevi o controle OL+CL para motor de passos.
Poderia fazer um teste sem ela por favor?

Reply:
de fato essa linha eu não sei o que ela ira resultar na lenta por pwm (acho que vou tirar ela do lenta por pwm), mas no código do motor de passo ela mantem o motor de passo aberto quando decaindo. tanto que ela já até estava no código do motor de passo, eu só coloquei ela no código do pwm, pensando que vai resultar na mesma coisa, mas eu não consigo testar pq meu carro é por motor de passo. 

olha ela ali no código 2023.05, ela mantem o motor de passo aberto em decaimento usando os números da tabela de marcha lenta, mas o problema é que esse código roda mesmo quando em CL uma coisa que não precisa pq não tem nem como editar a tabela de marcha lenta no tunerstudio quando em CL só quando em OL+CL.
![lenta pwm antiga](https://github.com/noisymime/speeduino/assets/143891251/f71cf403-9767-4590-8232-5e43479eb860)


sorry for speaking in another language

Reply:
Ah, fazem quase 3 anos que escrevi isso, se já tem lá pode deixar, desculpa a discussão, minha memória falhou.
Bom trabalho. Apenas abra a PR novamente por favor

Reply:
Vou abrir novamente, mas vou comentar a linha na parte de lenta por pwm, já que eu não consigo testa-la. 

Mas a lenta por motor de passo ficou bem melhor em desacelerações pq a válvula se mantem aberta pelo RPM, não só pelo TPS igual era antes, agora  ficou igual a da megasquirt.

Reply:
Não precisa alterar nada, eu realmente esqueci que tinha feito isso, pode deixar igual nos 2.
Desculpa mesmo, esqueci totalmente, faz muito tempo que eu fiz isso no motor de passos